### ISSUE TITLE
Mask-RCNN Segfault with TF2 trainer Inception Resnet V2 - Object Detection API

### ISSUE THREAD

**Issue by louisquinn** [tensorflow/models#10579]
# Prerequisites

Please answer the following questions for yourself before submitting an issue.

- [ y] I am using the latest TensorFlow Model Garden release and TensorFlow 2.
- [ y] I am reporting the issue to the correct repository. (Model Garden official or research directory)
- [y ] I checked to make sure that this issue has not already been filed.

## 1. The entire URL of the file you are using

https://github.com/tensorflow/models/tree/master/research/object_detection

## 2. Describe the bug

I'm attempting to retrain the pretrained Inception Resnet V2 model. THe main issue is the segfaults, however there are some config issues too. 

I have attempted this with other variants of feature extractors: (`faster_rcnn_resnet50_keras`) and same result.

Couple of issues here:

**1. Errors in the pretrained `pipeline.config`:**
* Random "s" character in `input_path: "PATH_TO_BE_CONFIGURED"s` in the `eval_input_reader`
* Missing `fine_tune_checkpoint_version: V2` in the `train_config`

**2. Segfault:**
* See below for the logs (truncated):

```
Number of filenames to read: 1
num_readers has been reduced to 1 to match input file shards.
From /var/sensen/envs/ai-platform/lib/python3.8/site-packages/object_detection/builders/dataset_builder.py:100: parallel_interleave (from tensorflow.python.data.experimental.ops.interleave_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use `tf.data.Dataset.interleave(map_func, cycle_length, block_length, num_parallel_calls=tf.data.AUTOTUNE)` instead. If sloppy execution is desired, use `tf.data.Options.deterministic`.
From /var/sensen/envs/ai-platform/lib/python3.8/site-packages/object_detection/builders/dataset_builder.py:235: DatasetV1.map_with_legacy_function (from tensorflow.python.data.ops.dataset_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use `tf.data.Dataset.map()
From /var/sensen/envs/ai-platform/lib/python3.8/site-packages/tensorflow/python/autograph/impl/api.py:458: calling map_fn (from tensorflow.python.ops.map_fn) with dtype is deprecated and will be removed in a future version.
Instructions for updating:
Use fn_output_signature instead
From /var/sensen/envs/ai-platform/lib/python3.8/site-packages/tensorflow/python/util/dispatch.py:1082: sparse_to_dense (from tensorflow.python.ops.sparse_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Create a `tf.sparse.SparseTensor` and use `tf.sparse.to_dense` instead.
From /var/sensen/envs/ai-platform/lib/python3.8/site-packages/tensorflow/python/util/dispatch.py:1082: to_float (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use `tf.cast` instead.

Process finished with exit code 137 (interrupted by signal 9: SIGKILL)
```

Could this be an issue with the `tfrecord`s? I have verified them and they look okay. The same records work with `DeepMAC`.

## 3. Steps to reproduce

Below is my `pipeline.config`. I have basically just modified the given config from the TF2 Model Zoo here: http://download.tensorflow.org/models/object_detection/tf2/20200711/mask_rcnn_inception_resnet_v2_1024x1024_coco17_gpu-8.tar.gz

```
model {
  faster_rcnn {
    number_of_stages: 3
    num_classes: 8
    image_resizer {
      fixed_shape_resizer {
        height: 512
        width: 512
      }
    }
    feature_extractor {
      type: "faster_rcnn_inception_resnet_v2_keras"
      first_stage_features_stride: 16
      batch_norm_trainable: true
    }
    first_stage_anchor_generator {
      grid_anchor_generator {
        height_stride: 16
        width_stride: 16
        scales: 0.25
        scales: 0.5
        scales: 1.0
        scales: 2.0
        aspect_ratios: 0.5
        aspect_ratios: 1.0
        aspect_ratios: 2.0
      }
    }
    first_stage_box_predictor_conv_hyperparams {
      op: CONV
      regularizer {
        l2_regularizer {
          weight: 0.0
        }
      }
      initializer {
        truncated_normal_initializer {
          stddev: 0.01
        }
      }
    }
    first_stage_nms_score_threshold: 0.0
    first_stage_nms_iou_threshold: 0.7
    first_stage_max_proposals: 100
    first_stage_localization_loss_weight: 2.0
    first_stage_objectness_loss_weight: 1.0
    initial_crop_size: 17
    maxpool_kernel_size: 1
    maxpool_stride: 1
    second_stage_box_predictor {
      mask_rcnn_box_predictor {
        fc_hyperparams {
          op: FC
          regularizer {
            l2_regularizer {
              weight: 0.0
            }
          }
          initializer {
            variance_scaling_initializer {
              factor: 1.0
              uniform: true
              mode: FAN_AVG
            }
          }
        }
        use_dropout: false
        dropout_keep_probability: 1.0
        conv_hyperparams {
          op: CONV
          regularizer {
            l2_regularizer {
              weight: 0.0
            }
          }
          initializer {
            truncated_normal_initializer {
              stddev: 0.01
            }
          }
        }
        predict_instance_masks: true
        mask_prediction_conv_depth: 0
        mask_height: 33
        mask_width: 33
        mask_prediction_num_conv_layers: 4
      }
    }
    second_stage_batch_size: 100
    second_stage_post_processing {
      batch_non_max_suppression {
        score_threshold: 0.0
        iou_threshold: 0.6
        max_detections_per_class: 100
        max_total_detections: 100
      }
      score_converter: SOFTMAX
    }
    second_stage_localization_loss_weight: 2.0
    second_stage_classification_loss_weight: 1.0
    second_stage_mask_prediction_loss_weight: 4.0
    resize_masks: false
  }
}
train_config {
  batch_size: 8
  data_augmentation_options {
    random_horizontal_flip {
    }
  }
  data_augmentation_options {
    random_adjust_brightness {
    }
  }
  data_augmentation_options {
    random_adjust_contrast {
    }
  }
  data_augmentation_options {
    random_adjust_saturation {
    }
  }
  optimizer {
    momentum_optimizer {
      learning_rate {
        cosine_decay_learning_rate {
          learning_rate_base: 0.008
          total_steps: 9700
          warmup_learning_rate: 0.0
          warmup_steps: 1000
        }
      }
      momentum_optimizer_value: 0.9
    }
    use_moving_average: false
  }
  gradient_clipping_by_norm: 10.0
  fine_tune_checkpoint: "path-to/mask_rcnn_inception_resnet_v2_1024x1024_coco17_gpu-8/checkpoint/ckpt-0"
  num_steps: 9700
  max_number_of_boxes: 100
  fine_tune_checkpoint_type: "detection"
  use_bfloat16: false
  fine_tune_checkpoint_version: V2
}
train_input_reader {
  label_map_path: "path-to/label_map.pbtxt"
  load_instance_masks: true
  tf_record_input_reader {
    input_path: "path-to/train*.tfrecord"
  }
  mask_type: PNG_MASKS
  max_number_of_boxes: 10000
}
eval_config {
  num_visualizations: 100
  num_examples: 23
  eval_interval_secs: 300
  metrics_set: "coco_detection_metrics"
  metrics_set: "coco_mask_metrics"
  metrics_set: "coco_mask_metrics"
  use_moving_averages: false
  eval_instance_masks: true
  max_num_boxes_to_visualize: 10000
  visualize_groundtruth_boxes: true
  include_metrics_per_category: false
  batch_size: 1
}
eval_input_reader {
  label_map_path: "path-to/label_map.pbtxt"
  shuffle: true
  num_epochs: 1
  load_instance_masks: true
  tf_record_input_reader {
    input_path: "path-to/val*.tfrecord"
  }
  mask_type: PNG_MASKS
  max_number_of_boxes: 10000
}
```

## 4. Expected behavior

Logs from above and a segfault.

## 5. Additional context

None

## 6. System information

- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): `Ubuntu 20.04`
- TensorFlow version (use command below): `tensorflow-gpu==2.8.0`
- Python version: `3.8`
- CUDA/cuDNN version: `CUDA-11.2` & `cuDNN 8`
- GPU model and memory: `RTX2070 8GB`

**Comment by louisquinn** [comment]
UPDATE: Turns out I was OOMing, but the error message wasn't getting printed in Pycharm :shrug: . Will re-open this if I find any issues. Thanks!
